precise_driver/include/precise_driver/pflex_device.h

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+#include <precise_driver/precise_tcp_interface.h>
+#include <precise_driver/queue.h>
+
+#include <iostream>
+#include <vector>
+#include <mutex>
+#include <thread>
+
+#include <geometry_msgs/Pose.h>
+
+namespace precise_driver
+{
+    struct Profile{
+        int speed;
+        int speed2;
+        int accel;
+        int decel;
+        double accel_ramp;
+        double decel_ramp;
+        int in_range;
+        int straight;
+    };
+
+    class PFlexDevice
+    {
+    public:
+        explicit PFlexDevice(std::shared_ptr<PreciseTCPInterface> connection,
+                    std::shared_ptr<PreciseTCPInterface> status_connection);
+        ~PFlexDevice();
+
+        void startMoveJThread();
+
+        bool init(const int& profile_no, const Profile& profile);
```

Comment:
Is there a reason to pass primitive types as references that I am not aware of? By value would be better:
* Spare indirection of the reference.
* Give the compiler the hint to optimize the call since nothing else uses the variable.
* Pass 32 instead of 64 bits?

This is done in all of the functions inside this class.